modules/drivers/redshift/test/metabase/driver/redshift_test.clj

```diff
@@ -16,3 +21,47 @@
   (is (= "com.amazon.redshift.jdbc.Driver"
          (.getName (class (jdbc-proxy/wrapped-driver (java.sql.DriverManager/getDriver "jdbc:redshift://host:5432/testdb")))))
       "Make sure we're using the correct driver for Redshift"))
+
+(defn- query->native [query]
+  (let [native-query (atom nil)]
+    (with-redefs [execute/prepared-statement (fn [_ _ sql _]
+                                               (reset! native-query sql)
+                                               (throw (Exception. "done")))]
+      (u/ignore-exceptions
+       (qp/process-query query))
+      @native-query)))
+
+;; TODO: Add executed-by and card-id and such to this
+(deftest remark-test
+  (let [expected (str/replace
+                  (str
+                   "-- /* partner: \"metabase\", {\"dashboard_id\":null,\"chart_id\":null,\"optional_user_id\":1000,\"optional_account_id\":null,"
+                   "\"filter_values\":{\"userid\":1,\"firstname\":[\"Rafael\",\"Robert\"]}} */"
+                   " Metabase:: userID: 1000 queryType: MBQL queryHash: cb83d4f6eedc250edb0f2c16f8d9a21e5d42f322ccece1494c8ef3d634581fe2\n"
+                   "SELECT \"%schema%\".\"test_data_users\".\"id\" AS \"id\","
+                   " \"%schema%\".\"test_data_users\".\"name\" AS \"name\","
+                   " \"%schema%\".\"test_data_users\".\"last_login\" AS \"last_login\""
+                   " FROM \"%schema%\".\"test_data_users\""
+                   " WHERE (\"%schema%\".\"test_data_users\".\"id\" = 1 OR \"%schema%\".\"test_data_users\".\"name\" = ?"
+                   " OR \"schema_170\".\"test_data_users\".\"name\" = ?)"
+                   " LIMIT 2000")
+                  "%schema%" rstest/session-schema-name)]
+   (mt/test-driver
+    :redshift
+    (is (= expected
+           (query->native
```

Comment:
You can use `mt/mbql-query` shorthand to generate most of this query for you:

```clj
(assoc
  (my/mbql-query users
    {:filter [:or [:= $id [:value 1 {:name "userid"}]]
                  [:= $name [:value "Rafael" {:name "firstname"}]]
                  [:= $name [:value "Robert" {:name "firstname"}]]]
     :limit 2000})
  :info {...})
```

would actually macroexpand to the exact same query you've written. Should save time